### PR TITLE
Add scan sum support for duration types to libcudf

### PR DESCRIPTION
**Pull request by davidwendt**
Depends on #9518 
Reference https://github.com/rapidsai/cudf/pull/9518#issuecomment-952310775

Add support for inclusive scan of duration types with SUM aggregation.
Also refactor the `scan_tests.cpp` to put rank tests into `rank_tests.cpp`, common utilities into `scan_tests.hpp`, and move the actual unit test logic out of `cudf::test` namespace.

### CODE REVIEW

**Comment by davidwendt** [comment]
rerun tests

File: cpp/src/reductions/scan/scan_inclusive.cu [hunk 1]
```diff
@@ -156,17 +158,10 @@ struct scan_functor<Op, cudf::string_view> {
...
-  {
-    return cudf::is_relationally_comparable<T, T>() && !cudf::is_dictionary<T>() &&
-           (std::is_same_v<Op, DeviceMin> || std::is_same_v<Op, DeviceMax>);
-  }
-
   template <typename T>
   static constexpr bool is_supported()
   {
-    return std::is_arithmetic_v<T> || is_min_max_supported<T>();
+    return std::is_invocable_v<Op, T, T> && !cudf::is_dictionary<T>();
```
Comment (jrhemstad) [inline cpp/src/reductions/scan/scan_inclusive.cu:164]: It really makes me happy that all that was needed here was to just change how `is_supported` was implemented and suddenly whole new type/op combinations are automatically enabled without any other work.

**Review by jrhemstad — APPROVED** [review]

File: cpp/src/reductions/scan/scan_inclusive.cu (hunk 1 above)
Comment (ttnghia) [reply cpp/src/reductions/scan/scan_inclusive.cu:164]: I'm curious, if `T` is dictionary then will `std::is_invocable_v<Op, T, T>` returns `true`? If not, we also don't need to check `!cudf::is_dictionary<T>()` anymore.

Comment (davidwendt) [reply cpp/src/reductions/scan/scan_inclusive.cu:164]: Yes, `std::is_invocable` will be true for dictionary types on certain operations but dictionary is purposely not supported for scan.

**Review by ttnghia — APPROVED** [review]

**Comment by jrhemstad** [comment]
@gpucibot merge

**Pull request merged**